### PR TITLE
Wiki: Add page for Benefits of Drasil

### LINKED ISSUE #3778: Create a Wiki Page for Benefits of Drasil
As discussed in #3772, we would like to create a wiki page showcasing examples of the benefits of Drasil over other methods of writing software.

As discussed in the PR, @BilalM04 is going to create the document, and add it to the sidebar under Misc.

A first example will be based on the PR, showing how changing a single line of code affected 175 files in the generated code.

### PR DESCRIPTION
**Pull request by BilalM04**
Created a wiki page for the benefits of Drasil to list examples that showcase the power of generative programming within Drasil.

Closes #3778

### CODE REVIEW

**Comment by BilalM04** [comment]
@B-rando1 feel free to make any changes as you see fit.

**Review by smiths — APPROVED** [review]
Thank you @BilalM04.  This exactly what I was looking for.  I like the positive tone for introductory blub.  I'll approve, but wait for @JacquesCarette to review before merging.

File: wiki/Benefits-of-Drasil.md
Comment (JacquesCarette) [inline wiki/Benefits-of-Drasil.md:3]: Let's not start pages with "Welcome" statements that are furthermore self-referential (i.e. 'this page', 'this wiki'). Please rephrase to remove these self-referential statements.

File: wiki/Benefits-of-Drasil.md
Comment (JacquesCarette) [inline wiki/Benefits-of-Drasil.md:7]: Please invert the 'story': put the benefits first, and how this was achieved second.

**Review by JacquesCarette — CHANGES_REQUESTED** [review]

File: wiki/Benefits-of-Drasil.md
Comment (JacquesCarette) [inline wiki/Benefits-of-Drasil.md:3]: Definitely better, but still in the passive voice.

Drasil's use of generative programming has a transformative impact on the development process. Here we collect examples demonstrating [rest the same]

File: wiki/Benefits-of-Drasil.md
Comment (JacquesCarette) [inline wiki/Benefits-of-Drasil.md:7]: **[Cosmetic changes made easy](../issues/3772)**: Our generated code was missing an empty line between the header comment block and the start of the code. A one-line change fixed this issue; at the time of this changes, this resulted in updates to 175 files, i.e. eliminating the need for 175 manual edits.

**Review by JacquesCarette — CHANGES_REQUESTED** [review]

**Review by JacquesCarette — APPROVED** [review]

**Pull request merged**
